### ISSUE TITLE
[HOLD for payment 2024-11-05] [$250] mWeb/Chrome -Chat - When adding emoji in multiline message composer is not completelly autoscrolled

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50174]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.44-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5043700&group_by=cases:section_id&group_order=asc&group_id=306201
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open the staging.new.expensify.com website
2. Open any chat
3. Start typing a multiline message until a scroller is displayed
4. In the next line, add an emoji
5. Verify the compose box is autoscrolled to the bottom and the emoji is not overlapped

## Expected Result:
When adding an emoji in a multiline message, the compose box should be scrolled to the bottom so the emoji is not overlapped

## Actual Result:
When the scrollbar is displayed in a multiline message and the user adds an emoji in a new line, the compose box is not autoscrolled to the bottom and the emoji is overlapped until the user scrolls down

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/cced6122-2541-4ada-b320-aa81abef2533

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021843658720204692310</li>
        <li>Upwork Job ID: 1843658720204692310</li>
        <li>Last Price Increase: 2024-10-08</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @OfstadC</details>

**Comment by lanitochka17** [comment]
@OfstadC  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by OfstadC** [comment]
I tested this end of last week and then got side tracked. Reviewing again

**Comment by OfstadC** [comment]
i believe this is ND Quality - posted in Slack to confirm

**Comment by Skalakid** [comment]
Hello, I'm Michał from SWM, an expert agency, and I will investigate this issue

**Comment by s77rt** [comment]
@OfstadC Please assign @Skalakid

**Comment by Skalakid** [comment]
Hello, I've investigated this topic, and two problems are causing this issue. 

The first is connected to our current scroll-into-view logic inside the Live Markdown library. When trying to insert the emoji from the modal, we add the emoji and the space after it to the input value. Because of this space, the target element for scrolling-into-view is the text node, which has a different line height than the emoji. To fix it, we can take the parent line element and scroll the view into it. Also, we can use the https://github.com/Expensify/App/issues/45885#issuecomment-2363453857 fix and add some bottom padding on the scroll, like 2px.


The second problem is connected to default text input behavior. By default, HTML input/text area elements don't include padding at the bottom when scrolling text into view. The same happens in the native RN TextInput component. Because of that, the input isn't scrolled fully to the bottom when inserting the new line and adding text there.

<details>
<summary>Video</summary>

https://github.com/user-attachments/assets/a952a2bc-e7e6-44e1-b8d1-f873538c9569

</details>


Because of that, I think we shouldn't care about scrolling the input fully to the bottom because it's the default input behavior. Instead, let's focus on scrolling input so emojis in the line don't overlap in every scenario. What are your thoughts on this?

**Comment by Skalakid** [comment]
Here is a [PR](https://github.com/Expensify/react-native-live-markdown/pull/514) with the fix

**Comment by s77rt** [comment]
PR still under review

**Comment by s77rt** [comment]
https://github.com/Expensify/react-native-live-markdown/pull/514 is merged. Waiting on the App PR for the upgrade

**Comment by s77rt** [comment]
@Skalakid Do you have an ETA for the E/App PR?

**Comment by Skalakid** [comment]
Hi, sorry I had to prioritize other tasks and didn't have time to create it. Creating the PR with a bump right now

**Comment by Skalakid** [comment]
Here is the [PR](https://github.com/Expensify/App/pull/51311)

**Comment by s77rt** [comment]
On the schedule 😁

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: n/a (Upstream bug and it seems to be a problem since day one)
- The offending PR has been commented on: n/a
- A discussion in #expensify-bugs has been started: Not needed
- Determine if we should create a regression test for this bug: No (low priority)

**Comment by OfstadC** [comment]
Payment Summary above looks good 👍

**Comment by JmillsExpensify** [comment]
$250 approved for @s77rt
